File: API/src/interfaces/repositories/IUser.repository.ts
import { FileDto } from "../../dtos/files/file.dto";
import { SignupDto } from "../../dtos/auth/signup.dto";
import { PagedList } from "../../helper/pagedList";
import { UserParams } from "../../params/user.params";

export interface IUserRepository {
  addPhoto(userId: string, fileDto: FileDto): Promise<FileDto | undefined>;
  getUserByEmail(email: string): Promise<any>;
  getUserByEmailAndRole(email: string, role: string): Promise<any>;
  getUserById(id: string): Promise<any>;
  getUsers(userParams: UserParams): Promise<PagedList<any>>;
  likeComment(user: any, commentId: string): Promise<any>;
  likePost(user: any, postId: string): Promise<any>;
  signupUser(signupDto: SignupDto): Promise<any>;
  unlikeComment(user: any, commentId: string): Promise<any>;
  unlikePost(user: any, postId: string): Promise<any>;
}

File: Webapi/Core/Domain/Repositories/IUserRepository.cs
using Domain.Entities;
using SharedKernel;
using SharedKernel.DTOs;
using SharedKernel.Params;

namespace Domain.Repositories;

public interface IUserRepository
{
    Task<User?> GetUserByIdAsync(Guid id);
    Task<PagedList<UserDto>> GetUsersAsync(UserParams userParams);
}
